src/app/_components/BlockList/useSubscribeBlocks.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+
+import { NakamotoBlock } from '@stacks/blockchain-api-client/src/generated/models';
+import { Block } from '@stacks/stacks-blockchain-api-types';
+
+import { useStacksWebSocketClient } from './useStacksWebSocketClient';
+
+interface Subscription {
+  unsubscribe(): Promise<void>;
+}
+
+export function useSubscribeBlocks(handleBlock: (block: NakamotoBlock) => any) {
+  const [subscription, setSubscription] = useState<Subscription>();
+  const stacksWebSocketClient = useStacksWebSocketClient();
```

Comment:
This could be an issue if the socket hook is used multiple times. In that case, we would want to use a context wrapper to hold on to the socket connection and keep adding subs to it

Reply:
Thanks for the suggestion. I added memoization and [unit test](https://github.com/hirosystems/explorer/blob/36ff9b2915221a9c4f444c0e23e690fea1d5a377/src/app/_components/BlockList/__tests__/useStacksWebSocketClient.test.tsx) to handle this issue.

Reply:
I don't think that unit test is testing the the hook being used in different components

Reply:
Just to be clear we don't need to change anything. This is fine for now, but I think we have to be really careful abou thow we use this hook while we don't have a way of maintaining a reference to the socket across components